include/open62541/plugin/securitypolicy.h

```diff
@@ -319,6 +322,14 @@ struct UA_SecurityPolicy {
                                                     const UA_ByteString newCertificate,
                                                     const UA_ByteString newPrivateKey);
 
+    /* Creates a PKCS #10 DER encoded certificate request signed with the server's
+     * private key. */
+    UA_StatusCode (*createSigningRequest)(UA_SecurityPolicy *securityPolicy,
+                                          const UA_String *subjectName,
```

Comment:
As long as this API is not widely used, I have a few wishes.

- Const input strings as a non-pointer value.
- Add a `UA_KeyValueMap` for additional parameters (as a pointer that can be NULL).
  That way we can add future parameters via named entries of the map without changing the API.
  The same trick is used in the EventLoop.

Happy to discuss.

Reply:
I added the KeyValueMap parameter.
SubjectName does not need to be passed and is otherwise extracted from the existing certificate. So I would leave it as a pointer.